File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/EdgeKeyOne.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * This entity represents the output of an edge delete impulse
 */
class EdgeKeyOne: Type() {

    fun source() { 
        add {"source" } 
    }

    fun edgeId() { 
        add {"edgeId" } 
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/EdgeKeyOne.py
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class EdgeKeyOne:
    """
    This entity represents the output of an edge delete impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def source(self) -> Uuid:
        return Uuid(self.dictionary.get("source"))
    @property
    def edge_id(self) -> Uuid:
        return Uuid(self.dictionary.get("edgeId"))
